### ISSUE TITLE
CanStart is always false in 1.3

### ISSUE THREAD

**Issue by tigerdyret** [OfficeDev/skype-ios-app-sdk-samples#64]
After upgrading to 1.3 the canStart in:

- (void)conversationHelper:(SfBConversationHelper *)conversationHelper
              videoService:(SfBVideoService *)videoService
         didChangeCanStart:(BOOL)canStart;

is always false.
This means that the app receives video from the Skype for Business on a laptop, but not the other way around. The self video view in the app is enabled and is working.

**Comment by tigerdyret** [comment]
![img_3056](https://user-images.githubusercontent.com/14999112/30984048-80892b4c-a48c-11e7-8adf-e9830a99a1b5.PNG)
I have isolated the bug.
It seems that the SDK gets in a faulty state if the meeting url has been re-used.
It works the first time, then if the user ends the meeting and re-joins it, the user video is not sent to the agent (I have re-created the bug in the sample Banking App for swift as well) and the preview-view remains hidden.

I have attached two screenshots of the app and the desktop app where you can see the issue.

![skype-desktop](https://user-images.githubusercontent.com/14999112/30984031-738e4c06-a48c-11e7-83e0-000a74ed8f18.png)

**Comment by 4brunu** [comment]
I was about to report the same issue.

**Comment by 4brunu** [comment]
```
if ([videoService canStart]) {

    NSError *error = nil;
    
    [videoService start:&error];
    
    if (error) {
        NSLog(@"videoService start: %@", error.localizedDescription);
    }
    
}

```

The following code printed the following log `videoService start: The operation couldn’t be completed. E_TooManyRequests (E2-3-5)`

**Comment by tigerdyret** [comment]
@jovillas Can you please label this as a bug and give us an ETA? This is unfortunately yet another critical bug and renders my app for danish municipalities almost useless.

**Comment by tigerdyret** [comment]
@jovillas et. al.: Our customer reports that they constantly experience audio, but no video.
Sometimes video is received by the desktop application, but the app does not receive video from the desktop application, sometimes it's the other way around.
Only very seldom will video work both ways :(

I have spent so much time bug tracking this library that I'm starting to get a bit tired (consultant, spent +100 hours debugging that I have to pay for myself...)

**Comment by marconealberto** [comment]
yeah, first time I tried the sdk I found the same exact problem. Did anyone solve this? I don't see the github issue section very active from their side.

**Comment by tigerdyret** [comment]
@marconealberto I have just shipped our app in production. We still experience the issue sometimes, but mostly on iOS 9.3.5 of which we luckily have discontinued our support for.

**Comment by tigerdyret** [comment]
We still experience this bug as well and it is so frustrating that nothing has been done about it.
If only the SDK had a "retry connect" method or something like that, we could work around it.

**Comment by marconealberto** [comment]
the really frustrating thing is that it can be reproduced just by running the example on two different devices, not some weird unreproducible bug. 
We fortunately found this at the beginning of the development, we won't be pushing this to our clients for sure.
Btw, is anyone from the Skype team reading this?

**Comment by OABsoftware** [comment]
By now we're almost 9 months later and still this bug is present.
Can we expect an updated version of the SDK anytime soon?

**Comment by marconealberto** [comment]
I honestly don't think they really care about the development of the SDK (last commit a year ago - no answers to the issues). I see this is more of a "we have a sort-of-working SDK and it's more than enough".

**Comment by OABsoftware** [comment]
Sad but true.

Nevertheless I found a workaround:

Apply the following observers:
            videoService?.addObserver(self, forKeyPath: "canStart", options: [.initial], context: &kvo)
            videoService?.addObserver(self, forKeyPath: "canSetPaused", options: [.initial], context: &kvo)
            selfVideo?.addObserver(self, forKeyPath: "isPaused", options: [.initial], context: &kvo)

Then use this in your observerValue method:
        case "canStart", "canSetPaused", "isPaused":
             let isPaused = (selfVideo?.isPaused == true)
             let canSetPaused = (videoService?.canSetPaused == true)
             let canStart = (videoService?.canStart == true)
             videoStart.isEnabled =  (isPaused && canSetPaused) || canStart

It works...

**Comment by marconealberto** [comment]
thank you! we never pursued the integration in the end, just because if there are other bugs in the future, we are on our own :(

**Comment by OABsoftware** [comment]
True...
May I ask which alternative you picked?

**Comment by codingcronus** [comment]
Yeah, it's a joke. Unfortunately on our behalf 👎 

We have ditched the Skype for Business SDK as well. I have spent countless (actually not countless, but +400 hours!) of debugging, tracing memory leaks, testing work arounds etc. and finally convinced our client that the SfB SDK is too unstable and non-supported.

I have now created a native WebRTC solution which works perfectly. I will share my library (NativeScript so it works cross platform for iOS, Android and Web) shortly.

**Comment by OABsoftware** [comment]
Cool!
